src/driver/sequencing/mod.rs

```diff
@@ -20,32 +20,32 @@ pub mod driver;
 pub trait SequencingSource {
     /// Returns the next payload attributes to be built (if any) on top of
     /// `unsafe_head`. If no attributes are ready to be built, returns `None`.
-    async fn get_next_attributes(
+    async fn get_next_attributes<U: JsonRpcClient>(
         &self,
         state: &Arc<RwLock<State>>,
+        provider: &Provider<U>,
```

Comment:
having it in the constructor is nicer, putting it in `get_next_attributes` makes the trait less general... but i guess you're doing this to avoid multiple owners? 

one option might be to just make `await_engine_sync` a separate util function with `state` and `provider` as inputs, called by the runner before constructing the sequencing driver